server/src/modules/rules/dtos/rules.dto.ts

```diff
@@ -19,6 +20,7 @@ export class RulesDto {
   manualCollectionName?: string;
   dataType: EPlexDataType;
   tautulliWatchedPercentOverride?: number;
+  notifications?: Notification[];
```

Comment:
Only the ID is used, passing the entire configuration object seems wrong

```suggestion
  notificationIds?: number[];
```

Reply:
Sure thing. I've removed RulesDto casting as part of [this draft PR](https://github.com/jorenn92/Maintainerr/pull/1698) which may make this refactor easier.